pkg/build/controller/factory/factory.go

```diff
@@ -112,7 +112,7 @@ func (factory *BuildControllerFactory) CreateDeleteController() controller.Runna
 		RetryManager: controller.NewQueueRetryManager(
 			queue,
 			cache.MetaNamespaceKeyFunc,
-			limitedLogAndRetry(factory.BuildUpdater, 30*time.Minute),
+			controller.RetryNever,
```

Comment:
I don't think we should have failed the build anyway if the pod can't be cleaned up (by definition) so this is probably ok.